hc5/src/main/java/feign/hc5/ApacheHttp5Client.java

```diff
@@ -135,7 +135,12 @@ ClassicHttpRequest toClassicHttpRequest(Request request, Request.Options options
         entity = new ByteArrayEntity(data, null);
       } else {
         final ContentType contentType = getContentType(request);
-        entity = new StringEntity(new String(data), contentType);
+        if (request.charset() != null) {
+          String content = new String(request.body(), request.charset());
+          entity = new StringEntity(content, contentType);
```

Comment:
can we move this line outside the if, and let the if block just handle the `String content` creation?!

Reply:
Done.